bitcoin/src/psbt/mod.rs

```diff
@@ -59,6 +59,48 @@ pub struct Psbt {
     pub outputs: Vec<Output>,
 }
 
+#[derive(Debug, Clone, PartialEq)]
+/// This error is returned when extracting a Transaction from a Psbt.
+pub enum ExtractTxError {
+    /// The Fee Rate (`satoshis / vBytes`) is too high
+    AbsurdFeeRate {
+        /// The Fee Rate (`satoshis / vBytes`)
+        fee_rate: FeeRate,
+        /// The extracted Transaction (use this to ignore the error)
+        transaction: Transaction,
+    },
+    /// One or more of the inputs lacks value information (witness_utxo or non_witness_utxo)
+    MissingInputValue {
+        /// The extracted Transaction (use this to ignore the error)
+        transaction: Transaction,
+    },
+    /// Input value is less than Output Value, and the transaction would be invalid.
+    SendingTooMuch {
+        /// The original Psbt is returned untouched.
+        psbt: Psbt,
+    },
+}
+
+impl fmt::Display for ExtractTxError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ExtractTxError::AbsurdFeeRate { fee_rate, .. } =>
+                write!(f, "An absurdly high fee rate of {}", fee_rate),
```

Comment:
tbh `sats / kwu` is going to confuse end users who see this error...

Even some developers will have a hard time at figuring out what it means.

As an internal representation, sats/kwu does make sense. but as a Display impl, there is not currently any traction for the `sats/kwu` unit of fee rate.

Just thought I'd put in my 2 cents.

Reply:
I modified FeeRate's Display impl to use sat/vbyte instead.

Originally the Display impl said "123 sat/kwu"